packages/@uppy/tus/src/index.js

```diff
@@ -196,9 +178,21 @@ module.exports = class Tus extends Plugin {
 
       const upload = new tus.Upload(file.data, optsTus)
       this.uploaders[file.id] = upload
-      this.uploaderEvents[file.id] = createEventTracker(this.uppy)
+      this.uploaderEvents[file.id] = new EventTracker(this.uppy)
+
+      const queuedRequest = this.requests.run(() => {
+        if (!file.isPaused) {
+          upload.start()
+        }
+        // Don't do anything here, the caller will take care of cancelling the upload itself
+        // using resetUploaderReferences(). This is because resetUploaderReferences() has to be
+        // called when this request is still in the queue, and has not been started yet, too. At
+        // that point this cancellation function is not going to be called.
+        return () => {}
```

Comment:
Question: so we have to call `this.resetUploaderReferences(file.id)` before `queuedRequest.abort()`?

Reply:
The order doesn't matter. I added some comments around the upload() function because it's super complex now ... resetUploaderReferences() must be called whenever an upload is completed or cancelled, queuedRequest.abort() whenever an upload is cancelled or paused, queuedRequest.done() whenever an upload is completed (either successfully or with error).